### PR TITLE
Replace deployment templates with structs

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Replaces all deployment files with go structs

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/resources/apiserver/deployment.go [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+}
+
+func getVolumes() []corev1.Volume {
+	return []corev1.Volume{
+		{
+			Name: resources.ApiserverTLSSecretName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  resources.ApiserverTLSSecretName,
+					DefaultMode: resources.Int32(256),
```
Comment (guusvw) [inline api/pkg/resources/apiserver/deployment.go]: This one could be extracted as const

File: api/pkg/resources/resources.go [hunk 2]
```diff
@@ -26,3 +26,6 @@ type RoleBindingCreator = func(data *TemplateData, existing *rbacv1.RoleBinding)
 
 // ClusterRoleBindingCreator defines an interface to create/update RBAC ClusterRoleBinding's
 type ClusterRoleBindingCreator = func(data *TemplateData, existing *rbacv1.ClusterRoleBinding) (*rbacv1.ClusterRoleBinding, error)
+
+// DeploymentCreator defines an interface to create/update Deployment's
+type DeploymentCreator = func(data *TemplateData, existing *appsv1.Deployment) (*appsv1.Deployment, error)
```
Comment (guusvw) [inline api/pkg/resources/resources.go:31]: :heart:

Comment (mrIncompetent) [reply api/pkg/resources/resources.go:31]: I tried following the go-way of imports here.
Child packages import's the parent but not vice versa

File: api/pkg/resources/apiserver/deployment.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/resources/apiserver/deployment.go]: Done with https://github.com/kubermatic/kubermatic/pull/1047/commits/1e16fc2835f2797e695d2bd2329f9dacc0857757

**Comment by mrIncompetent** [comment]
@guusvw A more general question:
Would you prefer a package for each individual component (apiserver, controller-manager, scheduler, etc.)?
- `pkg/resources/apiserver`
- `pkg/resources/controllermanager`
- `pkg/resources/scheduler`
- ...

or rather a single flat `resources` package which contain all the functions to create Deployments, ConfigMaps, etc.?

File: api/pkg/resources/test/load_files_test.go [hunk 3]
```diff
@@ -30,10 +35,6 @@ import (
 	kubefake "k8s.io/client-go/kubernetes/fake"
 )
 
-const (
-	masterResourcePath = "../../../../config/kubermatic/static/master/"
```
Comment (guusvw) [inline api/pkg/resources/test/load_files_test.go:34]: Man, I'm loving getting rid of these relative file path

**Review by guusvw — APPROVED** [review]
LGTM

**Pull request merged**